src/event-handler/index.ts

```diff
@@ -0,0 +1,85 @@
+import { EventEmitter } from 'events';
+import Ain from '../ain';
+import {
+  BlockchainEventTypes,
+  BlockFinalizedEventConfig,
+  EventConfigType,
+  ValueChangedEventConfig,
+} from '../types';
+import EventFilter from './event-filter';
+import EventChannelManager from './event-channel-manager';
+
+export default class EventHandler {
+  private readonly _filters: { [filterId: string]: EventFilter };
+  private readonly _eventEmitters: { [filterId: string]: EventEmitter };
+  private _ain: Ain;
+  private _eventChannelManager: EventChannelManager;
+
+  constructor(ain: Ain) {
+    this._filters = {};
+    this._eventEmitters = {};
+    this._ain = ain;
+    this._eventChannelManager = new EventChannelManager(ain, this);
+  }
+
+  async connect() {
+    await this._eventChannelManager.connect();
+  }
+
+  disconnect() {
+    this._eventChannelManager.disconnect();
+  }
+
+  emitEvent(filterId: string, payload: any) {
+    const targetEventEmitter = this._eventEmitters[filterId];
+    if (!targetEventEmitter) {
+      throw Error(`Can't find event emitter by filter id (${filterId})`);
+    }
+    targetEventEmitter.emit('data', payload);
+  }
+
+  emitError(filterId: string, errorMessage: string) {
+    const targetEventEmitter = this._eventEmitters[filterId];
+    if (!targetEventEmitter) {
+      throw Error(`Can't find event emitter by filter id (${filterId})`);
+    }
+    targetEventEmitter.emit('error', errorMessage);
+  }
+
+  subscribe(
+      eventType: 'BLOCK_FINALIZED',
+      config: BlockFinalizedEventConfig): EventEmitter;
+  subscribe(
+      eventType: 'VALUE_CHANGED',
+      config: ValueChangedEventConfig): EventEmitter;
+  subscribe(eventTypeStr: string, config: EventConfigType): EventEmitter {
+    if (!this._eventChannelManager.isConnected) {
+      throw Error(`Event channel is not connected! You must call ain.eh.connect() before using subscribe()`);
+    }
+    const eventType = BlockchainEventTypes[eventTypeStr] as BlockchainEventTypes;
```

Comment:
Can we just specify the type of eventTypeStr as BlockchainEventTypes and skip L59-62? It seems the same type-checking is also done in L81. Maybe then we can remove L49-54 as well?

Reply:
Thanks! Looks duplicated codes! Done